src/wallet/wallet.go

```diff
@@ -99,6 +101,35 @@ type Options struct {
 	CryptoType CryptoType // wallet encryption type, scrypt-chacha20poly1305 or sha256-xor.
 }
 
+type Receiver struct {
+	Address string `json:"address"`
+	Coins   string `json:"coins"`
+	Hours   string `json:"hours"`
+}
+
+type HoursSelection struct {
+	Type        string  `json:"type"`
+	Mode        string  `json:"mode"`
+	ShareFactor string `json:"share_factor"`
+}
+
+type AdvancedSpendRequest struct {
```

Comment:
these json structs are better off in the gui library, then convert to the struct used by wallet as argument